### PR TITLE
239 replace deepsource with open source components

### LINKED ISSUE #1305: Replace Deepsource with open source components
Following modules should be included in the CI-Pipeline:
- Bandit(Security-Checks)
- mypy (already included)
- prospector(wrapper for: pylint, pep8, pep257, pyflakes, mccabe, dodgy, pyroma, vulture) -> flake8 > pyflakes, mccabe includes flake8, 
- trivy (Security-Checks for the Repository)
- flake8

Whitelisting comments must be replaced with the ones from the new components.

### PR DESCRIPTION
**Pull request by ernstleierzopf**
# Make sure these boxes are signed before submitting your Pull Request -- thank you.

# Must haves
- [ ] I have read and followed the contributing guide lines at https://github.com/ait-aecid/logdata-anomaly-miner/wiki/Git-development-workflow
- [ ] Issues exist for this PR
- [ ] I added related issues using the "Fixes #<issue-id>"-notations
- [ ] This Pull-Requests merges into the "development"-branch

Fixes #1305 

# Submission specific

- [ ] This PR introduces breaking changes
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed

# Describe changes:
- Please disable/remove Deepsource  after accepting this PR.

### CODE REVIEW

**Review by whotwagner — CHANGES_REQUESTED** [review]

File: docs/images/aminer-config-color.drawio [hunk 1]
```diff
@@ -1 +1 @@
...TgfoH5/PLD7wJ/jReo/13HtJvH0w1zAGSXL2xelpeVAjEg6eyGBQEJyFiYAaWxxhztElhICuOQnphi/meiCtTFLc0Fy3DL9vaAIOfEZPWP3IuhkHKnpWCm44os7KTqlHPhRhHp4tvbNYchQMiukecFaKJ6uDOFXrle5atro+Nkq4yxS0RraofVM4T1UM3fEVBIf4F3BaYxt3FMAE644z+RCtKKDv3BqHnb4ULguUOE9Kye76xNBeauHrOVVwdRxPWK5x4giPM0nL12OGTZH97Z6ydT0Ez3aEz2sYItdp4c2y3dsaey+0c+R1zUvf8kT1owlv28qOHd72jTPF2deG2dc99N5ij7W507m9NqAcrsLyn4ey/L1hHN9i5tQWlv3bB2gaE/lYT2Pnjwbradyt36zojYAEwDNvQA9a2Hw/WG4D9hBLEoOy4oskORzmDHNNkwFtw4G9BTaiMrTo8lQkSHCUyjBPckLAvJVUsIt2FqtjjMJS3GVWx0flGsgDX6a7PYLg+m85YdKZLlPmIt7tE0g7xQRQjWXyFLMNJVNWIG7Uq36lavV324/mjR0XgPFpF/xoVFwkkxW8dL+toj0RL99WxeoC3hXw+iBd8gUlIfot5WUzv4WJauiOJN2JfbedeAKyRSOK+J1/tCYXLse9i4S81gfWvS9Roi89M7We1e0S+faz2yOZ8X6xXHCiqbOOOoNNCJnAFC5SEqrgOCMwyHHTpVu9jzsvGKo1sjuTB7LX81nxROEj/TtJt0RtbRCsbQ0SE8Ifu8GNI1R1eKRY3rsXzDP+pu4ybXk+XjOYsQKqf1cr8Tg5Vn4mqoThkEeKDoQR1WLSapbJB9isPbRmdrLQolGM2TlST/Q9p1bFY4bbT9XJaVVtqxXIff1A1hyfVV5yicwb/fcZVjPLy5AzWfhVnqfUAyPUW72GEFv0baozGXGd6czPte0Zj1nB3+5ykOX+nirYUFHGkVq1sqzcFR7S6r1TD3OaGszzgORMT0KA78YPHshj8D9JG/bl4A8XqTbGodqCBYM7o3/i0KeYN4MsgY6MuKeMxjWgCyXNj7WFq2nyhNFV6/UCcFwoezDntqolOmP8luz9Zjrr8u1W1PqmhzxdFJceF0Kgv09sTbjqEEjDOkcoV+0wZh0w2HLrD1dHWBXHFZfOn+DIuaf6hATz/Cw==</diagram></mxfile>
```
[the remark this answers is not in the document]
Comment (ernstleierzopf) [reply docs/images/aminer-config-color.drawio:1]: newline was added as a result of the pre-commit hook.
I ran the pre-commit on all files, so no issues will happen in future commits.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -2,28 +2,39 @@
 # See https://pre-commit.com/hooks.html for more hooks
 repos:
 -   repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v3.2.0
+    rev: v4.5.0
     hooks:
     -   id: trailing-whitespace
     -   id: end-of-file-fixer
     -   id: check-yaml
     -   id: check-added-large-files
```
[the remark this answers is not in the document]
Comment (ernstleierzopf) [reply .pre-commit-config.yaml:11]: done

File: aecid-testsuite/runFlake8.sh [hunk 3]
```diff
@@ -0,0 +1,2 @@
+python3 -m flake8 --max-line-length=140 --exclude=aecid-testsuite/ /usr/lib/logdata-anomaly-miner
```
[the remark this answers is not in the document]
Comment (ernstleierzopf) [reply aecid-testsuite/runFlake8.sh]: done

File: source/root/usr/lib/logdata-anomaly-miner/aminer.py [hunk 4]
```diff
@@ -677,7 +671,7 @@ def graceful_shutdown_handler(_signo, _stackFrame):
                     aminer_user_id = getpwnam(tmp_username).pw_uid
                 if tmp_group is not None:
                     aminer_group_id = getgrnam(tmp_group).gr_gid
-            except:  # skipcq: FLK-E722
+            except KeyError:
```
[the remark this answers is not in the document]
Comment (ernstleierzopf) [reply source/root/usr/lib/logdata-anomaly-miner/aminer.py:674]: This is the only exception I would expect. The error message (which is unchanged) only explains this error as the user or group name could not be resolved. Other errors will fail the aminer anyways and should either be returned as they are or they should be catched explicitly.

I see nothing that could go wrong with this approach, as the program quits anyways.

File: .pre-commit-config.yaml [hunk 5]
```diff
@@ -2,28 +2,39 @@
...
 -   repo: https://github.com/pre-commit/mirrors-mypy
-    rev: v0.790
+    rev: v1.9.0
     hooks:
     -   id: mypy
-        additional_dependencies: [pydantic]
+        additional_dependencies: [pydantic, types-PyYAML]
+        args: [--install-types, --ignore-missing-imports, --disable-error-code=attr-defined, --implicit-optional]
 
--   repo: https://gitlab.com/pycqa/flake8
```
[the remark this answers is not in the document]
Comment (ernstleierzopf) [reply .pre-commit-config.yaml:18]: I add this to the Jenkinsfile as an test which produces output, but it does not fail the pipeline. We did not fix the complexity thing and running the tests through would not be possible any more.

[the remark this answers is not in the document]
Comment (ernstleierzopf) [reply .pre-commit-config.yaml:18]: also checked and added other plugins from flake8.

**Review by whotwagner — APPROVED** [review]

**Pull request merged**